javascript/javascript1/week4/weeek4.js

```diff
@@ -0,0 +1,139 @@
+
+let name = "";
+let todos=[];
+
+function getReply(command) {
+  
+  if (command.includes("Hello my name is")) {
+    let newName = command.split("Hello my name is ")[1]; 
+    if (newName === name) {
+      return `You've already introduced yourself as ${name}`;
+    }
+    name = newName;
+    return `Nice to meet you ${name}`;
+  }
+
+  
+  if (command.includes("What is my name")) {
+    if (name) {
+      return `Your name is ${name}`;
+    } else {
+      return "I don't know your name yet.";
+    }
+  }
+  
+if (command.includes("Add") && command.includes("to my todo")) {
+    let todoItem = command.split("Add ")[1].split(" to my todo")[0]; 
+    todos.push(todoItem); 
+    return `${todoItem} added to your todo`;
+}
+
+if (command.includes("Remove") && command.includes("from my todo")) {
+  let todoItem = command.split("Remove ")[1].split(" from my todo")[0]; 
+  let index = todos.indexOf(todoItem); 
+  if (index > -1) {
+    todos.splice(index, 1); 
+    return `Removed ${todoItem} from your todo`;
+  } else {
+    return `${todoItem} is not in your todo`;
+  }
+}
+
+if (command.includes("What is on my todo")) {
+    if (todos.length === 0) {
+      return "Your todo list is empty";
+    } else {
+      return `You have ${todos.length} todos: ${todos.join(", ")}`;
+    }
+  }
+  
+  if (command.includes("What day is it today")) {
+    let today = new Date(); 
+    let day = today.getDate(); 
+    let monthNames = [
+      "January", "February", "March", "April", "May", "June", 
+      "July", "August", "September", "October", "November", "December"
+    ];
+    let month = monthNames[today.getMonth()]; 
+    let year = today.getFullYear(); 
+    return `${day}. of ${month} ${year}`; 
+  }
+
+  
+if (command.includes("What is")) {
+    let mathExpression = command.replace("What is", "").trim(); 
+    let parts = mathExpression.split(" "); 
+    let num1 = parseFloat(parts[0]); 
+    let operator = parts[1]; 
+    let num2 = parseFloat(parts[2]); 
+  
+    
+    if (operator === "+") {
+      return num1 + num2;
+    } else if (operator === "-") {
+      return num1 - num2;
+    } else if (operator === "*") {
+      return num1 * num2;
+    } else if (operator === "/") {
+      return num2 !== 0 ? num1 / num2 : "Cannot divide by zero";
+    } else {
+      return "I can only perform +, -, *, and / operations";

```

Comment:
Great error message. We frequently are guilty of writing error messages that say things like "Invalid input", or some variation on "I can't do that". This can be frustrating to the user - because from their perspective they have no idea what's actually wrong, or what else they can try.

Your error message instead helps the user to understand how the program _does_ work and gives clues about what the user can do about the error. It's much more helpful!